mooncake-store/src/client_buffer.cpp

```diff
@@ -27,6 +27,9 @@ std::shared_ptr<ClientBufferAllocator> ClientBufferAllocator::create(
 ClientBufferAllocator::ClientBufferAllocator(size_t size,
                                              const std::string& protocol)
     : protocol(protocol), buffer_size_(size) {
+    if (size == 0) {
+        return;
```

Comment:
In this early return case, let us ensure buffer_ and allocator_ are both properly initialized (to nullptr).